src/Symfony/Component/HttpKernel/HttpCache/Store.php

```diff
@@ -317,14 +317,30 @@ private function getMetadata($key)
     /**
      * Purges data for the given URL.
      *
+     * This method purges both the http and the https version of the cache entry.
+     *
      * @param string $url A URL
      *
-     * @return bool true if the URL exists and has been purged, false otherwise
+     * @return bool true if the URL exists with either http or https scheme and has been purged, false otherwise
      */
     public function purge($url)
     {
-        $key = $this->getCacheKey(Request::create($url));
+        $http = preg_replace('#^https#', 'http', $url);
```

Comment:
if $url is already http://, nothing changes.

this was previously 
```php
        $url2 = 0 === strpos($url, 'https://')
            ? preg_replace('#^https#', 'http', $url, 1)
            : preg_replace('#^http#', 'https', $url, 1);

        return $this->doPurge($url) || $this->doPurge($url2);
```
but i find this more readable. we trade one preg_replace for a strpos. given that we do multiple filesystem operations, the speed difference should not matter.